src/main/java/Duke.java

```diff
@@ -1,10 +1,200 @@
+import java.awt.*;
+import java.util.List;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+
 public class Duke {
+    private static List<Task> tasks = new ArrayList<>();
+
     public static void main(String[] args) {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        System.out.println(logo);
+
+        Scanner scanner = new Scanner(System.in);
+
+        try {
+            readFile();
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found");
+        }
+        greet();
+        while (scanner.hasNextLine()) {
+            String command = scanner.nextLine();
+            try {
+                handleCommand(command);
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+    }
+
+    private static void greet() {
+        System.out.println("Hello! I'm Duke\n What can I do for you?");
+    }
+
+    private static void exit() {
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+    private static void echo(String command) {
+        System.out.println("" + command);
+    }
+
+    private static void add(Task task) {
+        tasks.add(task);
+        System.out.println("Got it. I've added this task:\n" + task.toString()
+                + "\n Now you have " + tasks.size() + " tasks in the list.");
+    }
+
+    private static void done(Task task) {
+        task.setDone();
+        System.out.println("Nice! I've marked this task as done:\n" + task.toString());
+    }
+
+    private static void delete(int index) {
+        System.out.println("Noted. I've removed this task:\n" + tasks.get(index-1).toString());
+        tasks.remove(index-1);
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    }
+
+    private static void printList() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            if (tasks.get(i) != null) {
+                System.out.println((i + 1) + ". " + tasks.get(i).toString());
+            }
+        }
+    }
+
+    /**
+     * this method handles two types of error: invalid input (does not contain todo/deadline/event)
+     * and empty task (if the input starts with todo/deadline/event and the content is not empty,
+     * it is assumed that the input has correct structure)
+     */
```

Comment:
As per iP task on code quality, should the JavaDoc contain descriptions on the input parameter?